### ISSUE TITLE
Amazon Developer Publishing API Support

### ISSUE THREAD

**Issue by cpk1** [fastlane/fastlane#12376]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Is there any interest in supporting Amazon? They are finally rolling out their API: https://developer.amazon.com/developer-publishing
If there is interest in making this a part of Fastlane, how would one go about embarking on that journey? 😅 

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>N/A</pre>
</details>

### Environment

<details>
  <pre>N/A</pre>
</details>

**Comment by joshdholtz** [comment]
@cpk1 Thanks for bringing this to our attention! Issue #1963 had a discussion about this already and the conclusion looked like this should be added as a plugin - https://github.com/fastlane/fastlane/issues/1963#issuecomment-304130555

I am not sure yet if it should be an official fastlane plugin supported under the fastlane organization or not yet so let me get back to you on that 🙃

**Comment by lacostej** [comment]
I keep waiting on Amazon to offer a non restricted API. See #1963 for history

Their API is still restricted.
https://developer.amazon.com/docs/dev-publishing-api/devpubapi-create-and-submit-a-new-app.html#preconditions
